[DesSnowy] iP

DesSnowy opened the pull request:
### **Snowy**

> "Your mind is for having ideas, not holding them." - David Allen ([Source](https://dansilvestre.com/productivity-quotes))

Snowy is a 

- text-based
- easy to learn
- ~Fast~ SUPER FAST to use

All you need to do is,

1. download it.
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you. 😉

And its **FREE**!

Features:

- [X] Manage your task
- [ ]  Some unknown features

If you are a Java programmer, you can use it to practive Java too.
Here is the main method:
```  
public static void main(String[] args) {
    new Snowy("data/snowy.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

DesSnowy committed:
Rename Duke to Snowy

DesSnowy committed:
Add greeting message at start

DesSnowy committed:
Add exit statement

DesSnowy committed:
Add a method that repeat last command before exiting

DesSnowy committed:
Add echo function which stops when bye is typed

DesSnowy committed:
Add ability to add and list task

DesSnowy committed:
Add Task class

DesSnowy committed:
Change implementation to use Task class

DesSnowy committed:
Add ability to mark task as complete

DesSnowy committed:
Add ability to unmark a task

DesSnowy committed:
Add ToDo class

DesSnowy committed:
Add Deadline class

DesSnowy committed:
Add Event class

DesSnowy committed:
Change Snowy to use new task classes

DesSnowy committed:
Add  basic test cases

DesSnowy committed:
Add more test cases for errors

DesSnowy committed:
Adjust to solve spaces deleted at end of lines

DesSnowy committed:
Refactor to use arrayList as storage

DesSnowy committed:
Add ability to delete tasks

DesSnowy committed:
add new data file

DesSnowy committed:
Abstract out adding task to methods

DesSnowy committed:
Add ability to store tasks in harddrive

DesSnowy committed:
Merge branch 'branch-Level-7'

DesSnowy committed:
Add LocalDate to DeadLine and Event class

DesSnowy committed:
Change test cases to match new dates

DesSnowy committed:
Merge branch 'branch-Level-8'

DesSnowy committed:
Add Storage class

DesSnowy committed:
Add TaskList class

DesSnowy committed:
Add Ui class

DesSnowy committed:
Add integration across all files

DesSnowy committed:
Add ui when changing tasks status

DesSnowy committed:
Change test output to match new ui

DesSnowy committed:
Merge branch 'branch-MoreOOP'

DesSnowy committed:
Add all class to a package

DesSnowy committed:
Merge branch 'branch-Packages'

DesSnowy committed:
Merge branch 'add-gradle-support'

DesSnowy committed:
Add edge case for invalid date

DesSnowy committed:
Add JUnit test cases

DesSnowy committed:
Merge branch 'Branch-JUnit'

DesSnowy committed:
Rename file in build.gradle

DesSnowy committed:
Fix javaDoc

DesSnowy committed:
Fix coding standard

DesSnowy committed:
Add ability to find task

DesSnowy committed:
Add new test case

DesSnowy committed:
Change printing output of finding task

DesSnowy committed:
Merge branch 'branch-A-JavaDoc'

DesSnowy committed:
Merge branch 'branch-A-CodingStandard'

DesSnowy committed:
Resolve merge conflict in Snowy and Ui

DesSnowy committed:
Add checkstyle and fix styling

DesSnowy committed:
Add basic dialog view

DesSnowy committed:
change Ui to return string instead of printing

zackjh commented on src/main/java/snowy/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+        String temp = super.toFileStorage();
+        return String.format("D|%s|%s", temp, date.toString());
+    }
+}
```
Could `toFileStorage` be renamed to make it clearer?

I agree that `toFileStorage` serves a similar purpose as `toString`, but I also feel that `toFileStorage` is a little ambiguous.

zackjh commented on src/main/java/snowy/Storage.java:
```diff
@@ -0,0 +1,63 @@
+     */
+    public void save(String saveLines) throws SnowyException{
+        try{
+            FileWriter writer = new FileWriter(file);
```
Would `filewriter` be a better variable name than `writer`?

I think it might be a convention to name this variable `filewriter`, but I'm not entirely sure myself. I think it's worth looking into though.

zackjh commented on src/main/java/snowy/Task.java:
```diff
@@ -0,0 +1,52 @@
+abstract public class Task {
+    private boolean isCompleted = false;
+
+    public final DateTimeFormatter FORMATTER = DateTimeFormatter.ofLocalizedDate(FormatStyle.LONG);
```
Should `FORMATTER` be declared `public`?

zackjh commented on src/main/java/snowy/TaskList.java:
```diff
@@ -0,0 +1,243 @@
+            name = dataArray[2];
+            String dueDate = dataArray[3];
+            try {
+                Task newDateline = new Deadline(name, dueDate);
+                tasks.add(newDateline);
+                if (status.equals("1")) {
+                    newDateline.markComplete();
```
Should `newDateline` be `newDeadline` instead? I think it might be a typo - do check!

zackjh commented on src/main/java/snowy/TaskList.java:
```diff
@@ -0,0 +1,243 @@
+        Task newTask = new ToDo(description);
+        tasks.add(newTask);
+        System.out.println("New todo task added:\n" + newTask);
+    }
```
You could consider standardizing the casing for `ToDo` because in some places, it's `ToDo` (capital `D`) and in other places, it's `Todo` (small `d`).

zackjh reviewed (commented):
Things were named well in general!

I pointed out some minor things that you can consider refining but other than that, good job!

DesSnowy committed:
Add custom behaviour to DialogBox

DesSnowy committed:
Refactor to use fxml

DesSnowy committed:
Merge branch 'branch-Level-10'

munhuikim commented on src/main/java/snowy/Snowy.java:
```diff
@@ -0,0 +1,216 @@
+        }
+    }
+
+//    private void run() {
```
Could you consider removing unnecessary comments?

munhuikim commented on src/main/java/snowy/Task.java:
```diff
@@ -0,0 +1,52 @@
+ * Represents an abstract Task with all the methods that a Task should do.
+ */
+public abstract class Task {
+    private boolean isCompleted = false;
```
LGTM!

munhuikim commented on src/main/java/snowy/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+ * This class holds the deadline as a LocalDate.
+ */
+public class Deadline extends Task {
+    private LocalDate date;
```
Maybe consider changing the name to the endDate or deadline? It would be clearer.

munhuikim reviewed (commented):
In overall, LGTM 👍

thortol commented on src/main/java/snowy/Storage.java:
```diff
@@ -0,0 +1,63 @@
+     * Reads the data from the file and extracts the lines.
+     * @return An ArrayList where each element is a single line from the file.
+     * @throws SnowyException if the file does not exist and cannot be created.
+     * @throws SnowyException if the file exists but cannot be opened.
```
Should there be 2 @throws since load can only throw SnowyException?

thortol reviewed (commented):
LGTM! No major issues detected only minor nits. Great job!

DesSnowy committed:
Add assert calls to different methods

There are no calls to assert. This can cause the program to
experience error when assumptions do not hold true. This will
cause the program to crash without much clue to where the error
has taken place.

Adding in multiple assert lines allows us to check if the
assumptions made are true during run time. It also let us know
where the error was when the assumptions fails.

Let's add the assumptions made for different methods to be
checked using the assert method call.

DesSnowy committed:
Refactor code to improve code quality

The current code have multiple lines that are difficult to understand.
There are also variables that are declared but not used.

By removing unused variables and reorganizing the code, it will
make the code easier to understand.

DesSnowy committed:
Merge pull request #1 from DesSnowy/branch-A-Assertions

Add assert calls to different methods

DesSnowy committed:
Merge branch 'master' into branch-A-CodeQuality

DesSnowy committed:
Merge pull request #2 from DesSnowy/branch-A-CodeQuality

Refactor code to improve code quality

DesSnowy committed:
Add ability to change date of tasks

DesSnowy committed:
Merge branch 'branch-B-Snooze'

DesSnowy committed:
Improve styling in GUI

DesSnowy committed:
Change photo and add credits to the website

DesSnowy committed:
Add Ui.png

DesSnowy committed:
Add userGuide

DesSnowy committed:
Change README to include new features

DesSnowy committed:
Change README to include acknowledgement for GUI tutorial

DesSnowy committed:
Refactor getResponse in Snowy to apply SLAP

DesSnowy committed:
Add javadoc to more public methods

DesSnowy committed:
Fix bug which cause Snowy to fail if data directory is not found

DesSnowy committed:
Add ability to close application on bye

DesSnowy committed:
Improve code styling

DesSnowy committed:
Improve greeting string for better UX

DesSnowy committed:
Improve acknowledgement for pictures

DesSnowy committed:
Fix typo in README.md
